allowed.py

```diff
@@ -610,6 +618,17 @@ def check_file(filename: str, constructs: tuple) -> None:
         else:
             print(f"{filename}: can't parse: {message}")
 
+def read_jupyter_notebook(file_contents: str) -> str:
+    """Returns a string representation of all code cells in a Jupyter Notebook"""
+    jobject = json.loads(file_contents)
+    code_blocks = [x for x in jobject['cells'] if x['cell_type'] == 'code']
```

Comment:
Creating a list of all code cells uses extra memory. I'd suggest to add an if to the for-loop below:
```py
for cell in jobject...:
    if cell['cell_type'] == ...:
        for line in cell['source']:
            ...
```